core/[state|vm],trie: inspect and test getbalance costs for DNE addrs

meowsbits opened the pull request:
Benchmarks for DNE (Do Not Exist) address BALANCE calls, and some breadcrumbs for program flow.

```shell
> go test -bench Benchmark.*GetBalance ./core/state/...
goos: linux
goarch: amd64
pkg: github.com/ethereum/go-ethereum/core/state
BenchmarkStateDB_GetBalance_Existing-12          2528650
490 ns/op
BenchmarkStateDB_GetBalance_DNE-12                252471
4267 ns/op
PASS
ok      github.com/ethereum/go-ethereum/core/state      10.503s

> go test -bench Benchmark.*BalanceDNE ./core/vm/...
PASS
ok      github.com/ethereum/go-ethereum/core/vm 0.155s
goos: linux
goarch: amd64
pkg: github.com/ethereum/go-ethereum/core/vm/runtime
BenchmarkEVM_BalanceDNE_Gas40-12                   42105
28429 ns/op
BenchmarkEVM_BalanceDNE_Gas400-12                  43124
28241 ns/op
BenchmarkEVM_BalanceDNE_Gas4000-12                 30919
39121 ns/op
BenchmarkEVM_BalanceDNE_Gas40000-12                 9037
155781 ns/op
BenchmarkEVM_BalanceDNE_Gas004Gwei-12                 10
127698023 ns/op
BenchmarkEVM_BalanceDNE_Gas04Gwei-12                   1
1098864244 ns/op
BenchmarkEVM_BalanceDNE_Gas4Gwei-12                    1
11883739376 ns/op
BenchmarkEVM_BalanceDNE_Gas40Gwei-12                   1
119108679718 ns/op
PASS
ok      github.com/ethereum/go-ethereum/core/vm/runtime 139.523s
```

Signed-off-by: meows <b5c6@protonmail.com>

tzdybal commented:
I can reproduce error locally.
Looking at the code, expected error message is actually wrong (they are less detailed). We can change `!=` to `strings.HasPreffix`, or correct expected errors.

tzdybal commented:
The same error is visible on `release/1.9` on `ethereum/go-ethereum`. The code is updated, but test are old and fragile.

meowsbits committed:
core/[state|vm],trie: inspect and test getbalance costs for DNE addrs

> go test -bench Benchmark.*GetBalance ./core/state/...
goos: linux
goarch: amd64
pkg: github.com/ethereum/go-ethereum/core/state
BenchmarkStateDB_GetBalance_Existing-12          2528650
490 ns/op
BenchmarkStateDB_GetBalance_DNE-12                252471
4267 ns/op
PASS
ok      github.com/ethereum/go-ethereum/core/state      10.503s

> go test -bench Benchmark.*BalanceDNE ./core/vm/...
PASS
ok      github.com/ethereum/go-ethereum/core/vm 0.155s
goos: linux
goarch: amd64
pkg: github.com/ethereum/go-ethereum/core/vm/runtime
BenchmarkEVM_BalanceDNE_Gas40-12                   42105
28429 ns/op
BenchmarkEVM_BalanceDNE_Gas400-12                  43124
28241 ns/op
BenchmarkEVM_BalanceDNE_Gas4000-12                 30919
39121 ns/op
BenchmarkEVM_BalanceDNE_Gas40000-12                 9037
155781 ns/op
BenchmarkEVM_BalanceDNE_Gas004Gwei-12                 10
127698023 ns/op
BenchmarkEVM_BalanceDNE_Gas04Gwei-12                   1
1098864244 ns/op
BenchmarkEVM_BalanceDNE_Gas4Gwei-12                    1
11883739376 ns/op
BenchmarkEVM_BalanceDNE_Gas40Gwei-12                   1
119108679718 ns/op
PASS
ok      github.com/ethereum/go-ethereum/core/vm/runtime 139.523s

Signed-off-by: meows <b5c6@protonmail.com>
